### ISSUE TITLE
Missing dependency in yarn.lock

### ISSUE THREAD

**Issue by cpylua** [yarnpkg/yarn#6855]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
`media-typer` is missing in node_modules and yarn.lock.

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Start with a package.json file like this:
```
{
  "name": "yarn-bug",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "express": "^4.16.4",
    "webpack-serve": "^2.0.3"
  }
}
```
2. Run `yarn` to install packages
3. Run `find node_modules -name 'media-typer' -print`, it prints `node_modules/media-typer`
4. Run `grep media-typer yarn.lock`, it prints
```
media-typer@0.3.0:
  resolved "https://registry.yarnpkg.com/media-typer/-/media-typer-0.3.0.tgz#8710d7af0aa626f8fffa1ce00168545263255748"
```
5. Run `yarn add file-type`
6. Run `find node_modules -name 'media-typer' -print`, this time it prints nothing
7.  Run `grep media-typer yarn.lock`, prints nothing, too.

**What is the expected behavior?**

After step 5, `media-typer` should still be in `node_modules` and `yarn.lock`.

**Please mention your node.js, yarn and operating system version.**

yarn --version
1.12.3

node --version
v8.11.4

macOS Mojave version 10.14.2

**Comment by rally25rs** [comment]
I seem to be unable to reproduce this:

```
~/Projects/yarn-test 🐒   yarn add file-type
yarn add v1.12.3
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > webpack-serve@2.0.3" has unmet peer dependency "webpack@^4.0.0".
warning "webpack-serve > @webpack-contrib/cli-utils@1.0.2" has unmet peer dependency "webpack@^4.3.0".
warning "webpack-serve > @webpack-contrib/config-loader@1.2.1" has unmet peer dependency "webpack@^4.3.0".
warning "webpack-serve > @webpack-contrib/schema-utils@1.0.0-beta.0" has unmet peer dependency "webpack@^3.0.0 || ^4.0.0".
warning "webpack-serve > koa-webpack@5.1.1" has unmet peer dependency "webpack@^4.0.0".
warning "webpack-serve > time-fix-plugin@2.0.5" has unmet peer dependency "webpack@>=4.0.0".
warning "webpack-serve > koa-webpack > webpack-dev-middleware@3.5.0" has unmet peer dependency "webpack@^4.0.0".
warning "webpack-serve > koa-webpack > webpack-hot-client@4.1.1" has unmet peer dependency "webpack@^4.0.0".
[4/4] 📃  Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ file-type@10.7.0
info All dependencies
└─ file-type@10.7.0
✨  Done in 1.21s.

~/Projects/yarn-test 🐒   find node_modules -name 'media-typer' -print
node_modules/media-typer

~/Projects/yarn-test 🐒   grep media-typer yarn.lock
media-typer@0.3.0:
  resolved "https://registry.yarnpkg.com/media-typer/-/media-typer-0.3.0.tgz#8710d7af0aa626f8fffa1ce00168545263255748"
    media-typer "0.3.0"
```

**Comment by cpylua** [comment]
Solved with `yarn cache clean`. 

No idea what's going on here, it is 100% reproducible on my machine  before I clean the cache.
